rskj-core/src/test/java/co/rsk/peg/federation/FederationChangeIT.java

```diff
@@ -0,0 +1,420 @@
+                originalFederation = FederationFactory.buildNonStandardErpFederation(
+                    originalFederationArgs, erpPubKeys, activationDelay, activations);
+            case P2SH_ERP -> {
+                originalFederation = FederationFactory.buildP2shErpFederation(
+                    originalFederationArgs, erpPubKeys, activationDelay);
+            }
+            default -> throw new Exception(
+                String.format("Federation type %s is not supported", federationType));
+        }
+
+        // Set original federation
+        assertNotNull(originalFederation);
+        federationStorageProvider.setNewFederation(originalFederation);
+
+        // Set new UTXOs
+        var originalUTXOs = createRandomUTXOs(originalFederation.getAddress());
+        bridgeStorageAccessor.saveToRepository(
+            NEW_FEDERATION_BTC_UTXOS_KEY.getKey(), originalUTXOs, BridgeSerializationUtils::serializeUTXOList);
+
+         return originalFederation;
+    }  
+
+    private Federation commitPendingFederation(
+          List<Triple<BtcECKey, ECKey, ECKey>> federationKeys,
+          ActivationConfig.ForBlock activations) {
+        // Create Pending federation (doing this to avoid voting the pending Federation)
+        var newFederationMembers = federationKeys.stream()
+            .map(newFederatorKey ->
+                new FederationMember(
+                    newFederatorKey.getLeft(),
+                    newFederatorKey.getMiddle(),
+                    newFederatorKey.getRight()))
+            .toList();
+        var pendingFederation = new PendingFederation(newFederationMembers);
+        // Set pending federation
+        federationStorageProvider.setPendingFederation(pendingFederation);
+        federationStorageProvider.save(BRIDGE_CONSTANTS.getBtcParams(), activations);
+
+        // Proceed with the powpeg change
+        federationSupport.commitFederation(false, pendingFederation.getHash(), bridgeEventLogger);
+
+        // Since the proposed federation is commited, it should be null in storage
+        assertNull(federationStorageProvider.getPendingFederation());
+
+        // return the new federation
+        return pendingFederation.buildFederation(
+            Instant.EPOCH,
+            0,
+            BRIDGE_CONSTANTS.getFederationConstants(),
+            activations);
+    }
+
+    private void commitProposedFederation(ActivationConfig.ForBlock activations) {
+        // Verify that the proposed federation exists in storage
+        var proposedFederation = 
+            federationStorageProvider.getProposedFederation(BRIDGE_CONSTANTS.getFederationConstants(), activations);
+        assertTrue(proposedFederation.isPresent());
+
+        // As in commitPendingFederation util method, to avoid the SVP process
+        // we will commit directly
+        federationSupport.commitProposedFederation();
+    
+        // Since the proposed federation is commited, it should be null in storage
```

Comment:
```suggestion
        // Since the proposed federation is committed, it should be null in storage
```
just a little typo :)

Reply:
Thanks :)